src/codegeneration/ES6ClassTransformer.js

```diff
@@ -0,0 +1,262 @@
+   *
+   * @param {ClassExpression} tree
+   * @return {ParseTree}
+   */
+  transformClassExpression(tree) {
+    let {
+      elements,
+      initStaticVars,
+    } = this.transformClassElements_(tree);
+
+    let superClass = this.transformAny(tree.superClass);
+    let classExpression = new ClassExpression(tree.location, tree.name,
+        superClass, elements, tree.annotations, tree.typeParameters);
+
+    if (initStaticVars.length === 0) {
+      return classExpression;
+    }
+
+    this.pushTempScope();
+    let id = this.getTempIdentifier();
+    let className = createBindingIdentifier(id);
+    let statements = [
+      parseStatement `let ${id} = ${classExpression}`,
+      ...createStaticInitializerStatements(className, initStaticVars),
+      parseStatement `return ${className}`
+    ];
+    let body = createFunctionBody(statements);
+    this.popTempScope();
+
+    return createImmediatelyInvokedFunctionExpression(body);
+  }
+
+  getDefaultConstructor_(tree) {
+    if (tree.superClass) {
+      let param = createRestParameter(createIdentifierToken('args'));
+      let paramList = new FormalParameterList(null, [param]);
+      let body = createFunctionBody([parseStatement `super(...args)`]);
+      let name = createLiteralPropertyName(CONSTRUCTOR);
+      return new PropertyMethodAssignment(tree.location, false, null, name,
+          paramList, null, [], body);
+    }
+
+    return parsePropertyDefinition `constructor() {}`;
+  }
+
+  transformConstructorWithInitializer_(constructor, initExpression, superClass) {
+    if (superClass) {
+      let transformer = new SuperExpressionTransformer(initExpression);
+      return transformer.transformAny(constructor);
+    }
+
+    let statements = constructor.body.statements;
+    let initStatement = createExpressionStatement(initExpression);
+    statements = prependStatements(statements, initStatement);
+
+    return new PropertyMethodAssignment(constructor.location, false,
+        constructor.functionKind, constructor.name, constructor.parameterList,
+        constructor.typeAnnotation, constructor.annotations,
+        createFunctionBody(statements));
+  }
+}
+
+// TODO(vicb): Does not handle computed properties
+function createStaticInitializerStatements(className, initStaticMemberVars) {
+  return initStaticMemberVars.map((mv) => {
+    let propName = mv.name.literalToken.value;
+    return parseStatement
+        `Object.defineProperty(${className}, ${propName}, {enumerable: true,
+        configurable: true, value: ${mv.initializer}, writable: true})`;
+  });
+}
+
+// TODO(vicb): Does not handle computed properties
+function getInstanceInitExpression(initInstanceVars) {
+  let expressions = initInstanceVars.map((mv) => {
+    let name = mv.name.literalToken;
+    return parseExpression `this.${name} = ${mv.initializer}`;
+  });
+  return createCommaExpression(expressions);
+}
+
+export class SuperExpressionTransformer extends ParseTreeTransformer {
```

Comment:
Can you move this to its own file and rename it so that it is clear that this is used for member variables?